Sketcher: flaky GUI test TestExternalFacePreselection.testFacePickableUnderExternalTool

### Problem description

## Human part

I submitted a PR (https://github.com/FreeCAD/FreeCAD/pull/30919) for a crash I had encountered, and the PR failed one of the CI tests. I had claude code investigate, and it found that the test failure was unrelated to the PR. So I got github to just rerun the tests, and the failed one passed this time. It looks like the test is flaky (and has been tweaked a few times in the past for similar nondeterminsm), so below is the claude code (Opus 4.8) analysis of the failure and suggested fix. After I look at this a bit more, I'll file a PR.

## Description
`testFacePickableUnderExternalTool` (in `src/Mod/Sketcher/SketcherTests/TestExternalFacePreselection.py`) intermittently fails on Ubuntu CI ("FreeCAD GUI tests on install") with:

```
AssertionError: False is not true : Camera never framed the Pad face;
getPointOnScreen stayed at the viewport edge, so no valid interior hover point
could be computed.
```

Re-running the same job passes, so the failure is non-deterministic. It has bitten unrelated PRs (e.g. a Sketcher change that does not touch this code path), blocking them on a red check.

## Cause
The test calls `view.fitAll()` exactly once and then `wait_until(face_center_is_framed, ...)`. `wait_until` only pumps events; it never re-fits. In offscreen CI the 3D view's real size / aspect / device-pixel ratio is not always settled when `fitAll()` runs, so the camera frames against a not-yet-ready viewport and the face center stays at the viewport edge for the whole 5 s window.

The test was already hardened twice for related timing issues (commits "Make sure the frame is ready before checking for hover" and "harden Sketcher GUI probes for HiDPI"), but the one-shot `fitAll` remained.

## Suggested fix
Re-fit inside the wait loop so the framing self-corrects once the viewport is ready (and/or wait for a stable non-zero `getSize()` before the initial `fitAll`).

## Environment
Reproduces on Ubuntu CI offscreen rendering. Hard to reproduce on macOS (no offscreen GL context for the 3D view).


### Workbench affected?

None

### Steps to reproduce

Submit any PR. There's a small chance that this CI test will fail.

### Expected behavior

The test should not fail on a completely unrelated PR.

### Actual behavior

Sometimes the CI test randomly fails.

### Development version About Info (in Safe Mode)

```shell
The github CI test environment for Ubuntu.
```

### Last known good version (optional)

```shell

```

--- a/src/Mod/Sketcher/SketcherTests/TestExternalFacePreselection.py
+++ b/src/Mod/Sketcher/SketcherTests/TestExternalFacePreselection.py
@@ -227,6 +227,9 @@ class TestExternalFacePreselection(SketcherGuiTestCase):
             width, height = view.getSize()
             if width <= 0 or height <= 0:
                 return False
+            # Re-fit each probe: offscreen the viewport size/aspect may not be
+            # settled when fitAll() first runs, so fit until it frames the face.
+            view.fitAll()
             screen_x, screen_y = view.getPointOnScreen(face_center_3d)
             margin_x = width * 0.1
             margin_y = height * 0.1

